src/saleor_app/deps.py

```diff
@@ -48,9 +44,28 @@ async def saleor_token(
 
 
 async def verify_saleor_token(
-    domain=Depends(saleor_domain_header), token=Depends(saleor_token)
+    request: Request,
+    saleor_domain=Depends(saleor_domain_header),
+    token=Depends(saleor_token),
 ) -> bool:
-    is_valid = await verify_token(domain, token)
+    schema = "http" if request.app.use_insecure_saleor_http else "https"
+    async with get_client_for_app(
```

Comment:
I would not want to enforce a global error handler in the framework in case someone would want to deal with other GraphQL services in the app, but this brings an important topic, should those errors be HTTPErrors or should those be "framework" errors? What I'm getting at is that the framework should not enforce the error schema of an app but rather expose the ability to handle (documented) errors in the way a user of the framework wants, i.e. on `SaleorAuthenticationError` one would like to respond with a 403:
```
{
    "error": "Authentication with Saleor bad"
}
```
while another may want to do something like `application/problem+json` https://datatracker.ietf.org/doc/html/rfc7807

Reply:
I'll address that in #14